docs/modules/ROOT/pages/api/access.adoc

```diff
@@ -0,0 +1,487 @@
+```
+
+Contract module which provides a basic access control mechanism, where
+there is an account (an owner) that can be granted exclusive access to
+specific functions.
+
+This module provides an internal utility named `assert_only_owner`, which can be applied to your functions
+to restrict their use to the owner.
+
+[.contract-index]
+.Utilities
+--
+* xref:Ownable-assert_only_owner[`++InternalImpl::assert_only_owner(self)++`]
+--
+
+[.contract-index]
+.Functions
+--
+[.contract-subindex-inherited]
+.InternalImpl
+
+* xref:Ownable-initializer[`++initializer(self, owner)++`]
+* xref:Ownable-_transfer_ownership[`++_transfer_ownership(self, new_owner)++`]
+
+[.contract-subindex-inherited]
+.OwnableImpl
+
+* xref:Ownable-owner[`++owner(self)++`]
+* xref:Ownable-renounce_ownership[`++renounce_ownership(self)++`]
+* xref:Ownable-transfer_ownership[`++transfer_ownership(self, new_owner)++`]
+
+--
+
+[.contract-index]
+.Events
+--
+* xref:Ownable-OwnershipTransferred[`++OwnershipTransferred(previous_owner, new_owner)++`]
+--
+
+[#Ownable-Utilities]
+==== Utilities
+
+[.contract-item]
+[[Ownable-assert_only_owner]]
+==== `[.contract-item-name]#++assert_only_owner++#++(self: @ContractState)++` [.item-kind]#internal#
+
+Throws if called by any account other than the owner.
+
+[#Ownable-Functions]
+==== Functions
+
+[.contract-item]
+[[Ownable-initializer]]
+==== `[.contract-item-name]#++initializer++#++(ref self: ContractState, owner: ContractAddress)++` [.item-kind]#internal#
+
+Initializes the contract setting `owner` as the initial owner.
+
+Emits an xref:Ownable-OwnershipTransferred[OwnershipTransferred] event.
+
+[.contract-item]
+[[Ownable-_transfer_ownership]]
+==== `[.contract-item-name]#++_transfer_ownership++#++(ref self: ContractState, new_owner: ContractAddress)++` [.item-kind]#internal#
+
+Transfers ownership of the contract to a new account (`new_owner`).
+Internal function without access restriction.
+
+Emits an xref:Ownable-OwnershipTransferred[OwnershipTransferred] event.
+
+[.contract-item]
+[[Ownable-owner]]
+==== `[.contract-item-name]#++owner++#++(self: @ContractState) → ContractAddress++` [.item-kind]#external#
+
+Returns the address of the current owner.
+
+[.contract-item]
+[[Ownable-renounce_ownership--]]
+==== `[.contract-item-name]#++renounce_ownership++#++(ref self: ContractState)++` [.item-kind]#external#
+
+Leaves the contract without owner. It will not be possible to call
+`assert_only_owner` functions anymore. Can only be called by the current owner.
+
+NOTE: Renouncing ownership will leave the contract without an owner,
+thereby removing any functionality that is only available to the owner.
+
+[.contract-item]
+[[Ownable-transfer_ownership]]
+==== `[.contract-item-name]#++transfer_ownership++#++(ref self: ContractState, new_owner: ContractAddress)++` [.item-kind]#external#
+
+Transfers ownership of the contract to a new account (`new_owner`).
+Can only be called by the current owner.
```

Comment:
I don't think is necessary. Correct me if I'm wrong, but I think this would be relevant only if an Account implements the Ownable extension, and tries to transfer ownership, after setting the owner to 0, by calling the transfer_ownership internally in the execute method (only place where caller can be 0). I don't think we need to explain that trying to transfer the ownership in this case will revert.

Reply:
Haha my favorite edge case :) yes, that's a fair point

Reply:
wait why do we prevent calling from zero? is that the only scenario?

Reply:
> something is telling me this was my call...

Team effort! https://github.com/OpenZeppelin/cairo-contracts/pull/604#discussion_r1179863438

Reply:
at least i'm consistent